homeassistant/components/ondilo_ico/sensor.py

```diff
@@ -66,13 +65,16 @@ async def async_update_data():
         """
         try:
             pools = await hass.async_add_executor_job(api.get_pools)
+            for pool in pools:
```

Comment:
Can we do the iteration inside a function that we schedule on the executor so we don't switch context on every iteration?

Reply:
It would be good to do this iteration and the initial `api.get_pools` call in the same function and thread context.

Reply:
```py
def get_all_pool_data(api):
    """Fetch pools and add pool details and last measures to pool data."""
    pools = api.get_pools()
    for pool in pools:
        pool["ICO"] = api.get_ICO_details(pool["id"])
        pool["sensors"] = api.get_last_pool_measures(pool["id"])

    return pools

Reply:
Thanks @MartinHjelmare. Done.